### ISSUE TITLE
Custom script extension timeout

### ISSUE THREAD

**Issue by jermth** [Azure/custom-script-extension-linux#135]
Hi, 
I am puzzled by an CSE timeout. 

I am basically running the same custom script but in two different deployments: one in northcentralus, and the other in eastus (in different subscriptions).

The custom script in northcentralus timeouts after 5 mins, with the deployment in a `transitioning` state. The waagent.log has these lines in it:
```
2018/07/19 01:38:38.793526 WARNING [Microsoft.Azure.Extensions.CustomScript-2.0.6] [ExtensionError] Timeout(300): bin/custom-script-shim enable
[stdout]
Writing a placeholder status file indicating progress before forking: /var/lib/waagent/Microsoft.Azure.Extensions.CustomScript-2.0.6/status/0.status
```

The same script in eastus does not timeout and the waagent.log does not contain the lines above.

I have attached the 2 waagent logs. I believe this is related to other open issues (e.g [106](https://github.com/Azure/custom-script-extension-linux/issues/106)), but I'm not sure I understand what the solution is and why the 2 deployments differ.

The custom extension command invokes a python script:
```            
"resources": [
                {
                    "apiVersion": "[variables('computeApiVersion')]",
                    "dependsOn": [
                        "[variables('vmName')]"
                    ],
                    "location": "[resourceGroup().location]",
                    "name": "Install",
                    "properties": {
                        "autoUpgradeMinorVersion": true,
                        "protectedSettings": {
                            "commandToExecute": "[concat('python install.py', .......)]"
                        },
                        "publisher": "Microsoft.Azure.Extensions",
                        "settings": {
                            "fileUris": ["[variables('installScriptURL')]"]
                        },
                        "type": "CustomScript",
                        "typeHandlerVersion": "2.0"
                    },
                    "type": "extensions"
                }
            ],
```
[eastus.waagent.log](https://github.com/Azure/custom-script-extension-linux/files/2207942/eastus.waagent.log)
[northcentralus.waagent.log](https://github.com/Azure/custom-script-extension-linux/files/2207943/northcentralus.waagent.log)

**Comment by boumenot** [comment]
There was an issue in the [Linux agent](http://github.com/Azure/WALinuxAgent) that caused issues in a small percentage of VMs.  The version with the bug was 2.2.27 (and maybe 2.2.28), which was released to West Central US and North Central US.  A fix was made in 2.2.29, and that version is rolling out now.  If you are still experiencing the issue even with the latest version installed (2.2.29) please re-open.

**Comment by szaher** [comment]
I am facing the same issue on RHEL 7.6 instance
"customData": "echo 'IyEvYmluL2Jhc2gKc3VkbyB5dW0gdXBkYXRlIC15CnN1ZG8geXVtIGluc3RhbGwgLXkgamF2YS0x\nLjguMC1vcGVuamRrLWRldmVsCmN1cmwgLS1zaWxlbnQgLS1sb2NhdGlvbiBodHRwOi8vcGtnLmpl\nbmtpbnMtY2kub3JnL3JlZGhhdC1zdGFibGUvamVua2lucy5yZXBvIHwgc3VkbyB0ZWUgL2V0Yy95\ndW0ucmVwb3MuZC9qZW5raW5zLnJlcG8Kc3VkbyBycG0gLS1pbXBvcnQgaHR0cHM6Ly9qZW5raW5z\nLWNpLm9yZy9yZWRoYXQvamVua2lucy1jaS5vcmcua2V5CnN1ZG8geXVtIGluc3RhbGwgLXkgamVu\na2lucwpzdWRvIHN5c3RlbWN0bCBlbmFibGUgamVua2lucwpzdWRvIHN5c3RlbWN0bCBzdGFydCBq\nZW5raW5zCnN1ZG8gZmlyZXdhbGwtY21kIC0tcGVybWFuZW50IC0tem9uZT1wdWJsaWMgLS1hZGQt\ncG9ydD04MDgwL3RjcApzdWRvIGZpcmV3YWxsLWNtZCAtLXJlbG9hZAp3Z2V0IGh0dHBzOi8vcmVs\nZWFzZXMuaGFzaGljb3JwLmNvbS9wYWNrZXIvMS4zLjUvcGFja2VyXzEuMy41X2xpbnV4X2FtZDY0\nLnppcAp1bnppcCBwYWNrZXJfMS4zLjVfbGludXhfYW1kNjQuemlwCnN1ZG8gbXYgLi9wYWNrZXIg\nL3Vzci9iaW4v' | base64 -d > install.sh && sh install.sh"
{
  "type": "Microsoft.Compute/virtualMachines/extensions",
  "name": "[concat(variables('vm_name'),'/','extensions')]",
  "apiVersion": "2017-03-30",
  "location": "[resourceGroup().location]",
  "dependsOn": [
    "[concat('Microsoft.Compute/virtualMachines/', variables('vm_name'))]"
  ],
  "properties": {
    "publisher": "Microsoft.Azure.Extensions",
    "type": "CustomScript",
    "typeHandlerVersion": "2.0",
    "autoUpgradeMinorVersion": true,
    "settings": {
      "commandToExecute": "[variables('customData')]"
    },
    "protectedSettings": {}
}
}

it only does the yum -y update and stops !!
not sure what can cause that but the agent

waagent --version
WALinuxAgent-2.2.32.2 running on redhat 7.6
Python: 2.7.5
Goal state agent: 2.2.37
